pkg/controllers/clusterresourceplacementwatcher/watcher_integration_test.go

```diff
@@ -58,33 +58,19 @@ var _ = Describe("Test ClusterResourcePlacement Watcher", func() {
 		By("By creating a new clusterResourcePlacement")
 		crp := clusterResourcePlacementForTest()
 		Expect(k8sClient.Create(ctx, crp)).Should(Succeed())
-	})
-
-	Context("When creating new clusterResourcePlacement", func() {
-		AfterEach(func() {
-			By("By deleting crp")
-			createdCRP = clusterResourcePlacementForTest()
-			Expect(k8sClient.Delete(ctx, createdCRP)).Should(Succeed())
 
-			By("By checking crp")
-			Eventually(func() bool {
-				return errors.IsNotFound(k8sClient.Get(ctx, types.NamespacedName{Name: testCRPName}, createdCRP))
-			}, eventuallyTimeout, interval).Should(BeTrue(), "crp should be deleted")
-		})
+		By("By checking the placement queue before resetting")
+		// The event could arrive after the resetting, which causes the flakiness.
+		// It makes sure the queue is clear before proceed.
+		Eventually(func() bool {
+			return fakePlacementController.Key() == testCRPName
+		}, eventuallyTimeout, interval).Should(BeTrue(), "placementController should receive the CRP name when creating CRP")
 
-		It("Should enqueue the event", func() {
-			By("By checking placement controller queue")
-			Eventually(func() bool {
-				return fakePlacementController.Key() == testCRPName
-			}, eventuallyTimeout, interval).Should(BeTrue(), "placementController should receive the CRP name")
-		})
+		By("By resetting the placement queue")
+		fakePlacementController.ResetQueue()
 	})
 
 	Context("When updating clusterResourcePlacement", func() {
-		BeforeEach(func() {
-			By("By resetting the placement queue")
-			fakePlacementController.ResetQueue()
-		})
 		AfterEach(func() {
 			By("By deleting crp")
 			createdCRP = clusterResourcePlacementForTest()
```

Comment:
why creating this again?

Reply:
oh, it's not creating a new CRP by sending to the API server and it's building a CRP structure. the name is not good. Will rename this variable.